src/main/java/seedu/address/model/student/Student.java

```diff
@@ -291,4 +293,35 @@ public String studentEssentialPrinter() {
                 .append(getAcademicYear());
         return builder.toString();
     }
+
+    /**
+     * Returns whether a student is able to attend a training at a particular DateTime
+     */
+    public boolean isAvailableAtDateTime(LocalDateTime dateTime) {
+        DayOfWeek day = dateTime.getDayOfWeek();
+
+        LocalTime queryTime = LocalTime.from(dateTime);
+        LocalTime studentTime;
+
+        switch (day) {
+        case MONDAY:
+            studentTime = mondayDismissal.dismissalTime;
+            return studentTime.equals(queryTime) || studentTime.isBefore(queryTime);
+        case TUESDAY:
+            studentTime = tuesdayDismissal.dismissalTime;
+            return studentTime.equals(queryTime) || studentTime.isBefore(queryTime);
+        case WEDNESDAY:
+            studentTime = wednesdayDismissal.dismissalTime;
+            return studentTime.equals(queryTime) || studentTime.isBefore(queryTime);
+        case THURSDAY:
+            studentTime = thursdayDismissal.dismissalTime;
+            return studentTime.equals(queryTime) || studentTime.isBefore(queryTime);
+        case FRIDAY:
+            studentTime = fridayDismissal.dismissalTime;
+            return studentTime.equals(queryTime) || studentTime.isBefore(queryTime);
+        default:
+            return false;
```

Comment:
shouldnt this be true, since if the training is scheduled on saturdays and sundays, student will always be able to make it?